modules/core/src/test/scala/sangria/federation/FederationSpec.scala

```diff
@@ -0,0 +1,144 @@
+package sangria.federation
+
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import sangria.util.FutureResultSupport
+import sangria.macros._
+import sangria.execution.{ExecutionScheme, Executor}
+import sangria.marshalling.InputUnmarshaller
+import sangria.parser.QueryParser
+import sangria.schema.{AdditionalTypes, Argument, AstSchemaBuilder, FieldResolver, Schema}
+import sangria.validation.Violation
+
+import scala.util.Success
+
+class FederationSpec extends AnyWordSpec with Matchers with FutureResultSupport {
+  import FederationSpec._
+
+  "To support Apollo Federation, an unmarshaller" should {
+    "accept a map value as a scalar" in {
+      val AST =
+        graphql"""
+           schema {
+             query: Query
+           }
+
+           type Query {
+             entities(representation: _Any!): _Any
+           }
+
+            union _Entity = State
+
+            type State {
+             id: Int!
+             value: String!
+           }
+         """
+
+      val federationSupport =
+        new FederationSupport(sangria.marshalling.json4s.jackson.Json4sJacksonInputUnmarshaller)
+
+      val schema: Schema[Any, Any] = Schema.buildFromAst(
+        AST,
+        AstSchemaBuilder.resolverBased[Any](
+          FieldResolver.map(
+            "Query" -> Map(
+              "entities" -> (ctx => ctx.withArgs(federationSupport.representationArg)(arg => arg))
+            )
+          ),
+          AdditionalTypes(federationSupport._Any.Type)
+        )
+      )
+
+      val Success(query) = QueryParser.parse("""
+        query FetchState($representation: _Any!) {
+          entities(representation: $representation)
+        }
+        """)
+
+      val args: org.json4s.JValue = org.json4s.jackson.JsonMethods
+        .parse(""" { "representation": { "__typename": "State", "id": 1 } } """)
+
+      val result = Executor
+        .execute(schema = schema, queryAst = query, variables = args)(
+          executionContext = scala.concurrent.ExecutionContext.global,
+          marshaller = sangria.marshalling.queryAst.queryAstResultMarshaller,
+          um = federationSupport.inputMarshallerForFederation,
+          scheme = ExecutionScheme.Default
+        )
+        .await
+
+      result.renderPretty should be("""{
+        |  data: {
+        |    entities: "{\"__typename\":\"State\",\"id\":1}"
+        |  }
+        |}""".stripMargin)
+    }
+  }
+}
+
+object FederationSpec {
+
+  class FederationSupport[Node](default: InputUnmarshaller[Node]) {
+
+    trait NodeObject {
```

Comment:
I think we need some material than just the render method, I am working on refining the test because at the end we wanna be able to extract the information about an entity's keys from _Any